lib/SPIRV/SPIRVReader.cpp

```diff
@@ -53,6 +53,7 @@
 #include "VectorComputeUtil.h"
 
 #include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/SmallSet.h"
```

Comment:
I wonder if using small sets instead of vectors is a good idea (applicable both to SPIRVReader and SPIRVWriter). This does result in less verbose code (a simple `SmallSet<>::insert()` instead of `std::find()` + subsequent iterator checks). The doubts arise from the fact that these collections have a very low number of elements, so a set could impact the performance. I know that some optimizations take place for small numbers of elements in `SmallSet`, but not the nature of these.

Anyway, if there's any input on reviewers' side here, I would appreciate it.

Reply:
According to https://llvm.org/doxygen/SmallSet_8h_source.html#l00134 `SmallSet` uses vector under the hood if number of elements is less then template parameter `N`. `N` = 4 in your case.

So, I think it is just fine to use `SmalSet` in this case.